### ISSUE TITLE
haas-unraid is spinning up spun down drives every scan interval

### ISSUE THREAD

**Issue by chilman408** [IDmedia/hass-unraid#23]
Is there a way to prevent hass-unraid to poll drive that are spun down?

I'm trying to optimize power usage on my unraid and found this docker app is spinning up my driver every scan interval

**Comment by IDmedia** [comment]
Are you sure hass-unraid is waking up your drives? I use this and have no issues. Drives never spin up because of the plugin on the two unRAID servers I have. The solution was specifically designed to not spin up the drives.

**Comment by chilman408** [comment]
Yes, it is spinning up my array.  I disabled all of my dockers and VMs one by one.  This docker was the culprit for spinning up drives at an interval

**Comment by thedotor** [comment]
Same issue, I am running xfs array.

**Comment by thedotor** [comment]
Version 7.0.1

Have been running this for a few hours since the update you pushed. 
The drives are not spinning up now.

**Comment by genem2** [comment]
(Not sure if I should comment on a closed issue or not...)
Just learned of this component, and like all the sensors.
But, I have the drives spinning up immediatly, untill I stop the container.
```
May  3 18:35:42 ed800 emhttpd: spinning down /dev/sdd
May  3 18:35:42 ed800 emhttpd: spinning down /dev/sdb
May  3 18:35:43 ed800 emhttpd: read SMART /dev/sdb
May  3 18:35:45 ed800 emhttpd: read SMART /dev/sdd
```

I went through this recently, and worked out a script to only attempt smartctl for spinning drives:

```bash
# --- Disk Temperatures (only configured disks) ---
json_payload="{"

# --- Disk Temperatures (always include all disks) ---
declare -A disk_temps  # Store temps for all disks
for dev in /dev/sd?; do
  dev_name=$(basename "$dev")
  state=$(hdparm -C "$dev" 2>/dev/null | grep 'drive state is' | awk '{print $4}')
  if [[ "$state" == "active/idle" ]]; then
    temp=$(smartctl -a "$dev" | grep -iE 'Temperature_Celsius|Airflow_Temperature_Cel' | awk '{print $10}' | grep -oE '[0-9]+' | head -1)
    disk_temps["$dev_name"]=${temp:-null}
  else
    disk_temps["$dev_name"]=null
  fi
done

# Add all disk temps to JSON (even if null)
for dev_name in "${!disk_temps[@]}"; do
  json_payload+="\"disk_${dev_name}_temperature\":${disk_temps[$dev_name]},"
done
```

unRAID 7.0.1
HA Core  2025.4.4

**Comment by IDmedia** [comment]
It's better to make a new issue so it's easier to keep track.

I've never had issues with drives in my main array spinning up unnecessarily. I use this on two servers and haven't noticed any signs of the integration waking up drives. Can't speak for pools, though, as they rarely stay idle.
